cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/adapter/PluginInstantiator.java

```diff
@@ -114,6 +142,12 @@ public ClassLoader getArtifactClassLoader(ArtifactDescriptor artifactDescriptor)
     return (Class<T>) getArtifactClassLoader(artifactDescriptor).loadClass(pluginClass.getClassName());
   }
 
+  @SuppressWarnings("unchecked")
+  public <T> Class<T> loadClass(Location location,
```

Comment:
We still use ``Id.Artifact`` in ``ArtifactDescriptor`` to construct the path to the artifact jar.